packages/core/test/unit/WriteApi.test.ts

```diff
@@ -209,6 +210,41 @@ describe('WriteApi', () => {
       expect(writeOptions.writeFailed).to.not.throw()
     })
   })
+  describe('convert point time to line protocol', () => {
+    const writeAPI = createApi(ORG, BUCKET, 'ms', {
+      retryJitter: 0,
+    }) as PointSettings
+    it('converts empty string to no timestamp', () => {
+      const p = new Point('a').floatField('b', 1).timestamp('')
+      expect(p.toLineProtocol(writeAPI)).equals('a b=1')
+    })
+    it('converts number to timestamp', () => {
+      const p = new Point('a').floatField('b', 1).timestamp(1.2)
+      expect(p.toLineProtocol(writeAPI)).equals('a b=1 1')
+    })
+    it('converts number to timestamp', () => {
+      const d = new Date()
+      const p = new Point('a').floatField('b', 1).timestamp(d)
+      expect(p.toLineProtocol(writeAPI)).equals(`a b=1 ${d.getTime()}`)
+    })
+    it('converts undefined to local timestamp', () => {
+      const p = new Point('a').floatField('b', 1)
+      expect(p.toLineProtocol(writeAPI)).satisfies((x: string) => {
+        return x.startsWith('a b=1')
+      }, `does not start with 'a b=1'`)
+      expect(p.toLineProtocol(writeAPI)).satisfies((x: string) => {
+        return Date.now() - Number.parseInt(x.substring('a b=1 '.length)) < 1000
+      })
+    })
+    it('converts any value to a timestamp using value.toString', () => {
+      const p = new Point('a').floatField('b', 1).timestamp({
+        toString() {
+          return '123'
+        },
+      })
+      expect(p.toLineProtocol(writeAPI)).equals('a b=1 123')
+    })
```

Comment:
I actually recommend removing this functionality.  While it is generally a  good idea to be permissive in what you accept, calling `Number(x.toString())` (effectively) is almost certainly going to result in end users getting bugs in their software that *could* have been caught by the compiler.

This code *really* wants a `Date`, but since JS `Date`s only hold down to milliseconds it must also accept a number.  However, since JS numbers cannot accurately hold anything bigger that `2^53` it has to accept a `string` until such time as you stop supporting old browsers/NodeJS versions that are pre-bigint.

I think extending this even further to accept "anything that can turn into a string" (which is basically everything), and then having the user be surprised when their Uint8Array that is holding a 256-bit twos-complement encoded number doesn't convert properly, it would be better to just give them a compiler error that they can fix (by calling `.toString()` if that is really what they want).

Reply:
I tyhink that you did the same observations as I did year ago when I was introducing the Point. Now I was obsessed with a native `bigint` that I, unfortunately, can't realize in the function signature. I used `unknown` in place of it as a short path. After a few days, doing 2 steps back to take a look at the result, `unknown` is even worse than nothing. I'll remove it.